### ISSUE TITLE
Inconsistent errors when Latitude is 90 vs. -90

### ISSUE THREAD

**Issue by eliottgray** [mapbox/mercantile#106]
**TL;DR**: When Latitude is 90, resulting exception should be nearly identical to when Latitude is -90.

When calling mercantile.tile() when latitude is -90, a descriptive stack trace and error is raised:

```
>>> mercantile.tile(30, -90, 10, truncate=True)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/mercantile/__init__.py", line 307, in _xy
    y = 0.5 - 0.25 * math.log((1.0 + sinlat) / (1.0 - sinlat)) / math.pi
ValueError: math domain error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/mercantile/__init__.py", line 331, in tile
    x, y = _xy(lng, lat, truncate=truncate)
  File "/usr/local/lib/python3.7/site-packages/mercantile/__init__.py", line 309, in _xy
    raise InvalidLatitudeError("Y can not be computed: lat={!r}".format(lat))
mercantile.InvalidLatitudeError: Y can not be computed: lat=-90
>>> 
```

However, when latitude is 90, a different (and less descriptive) error is raised:
```
>>> mercantile.tile(30, 90, 10, truncate=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/mercantile/__init__.py", line 331, in tile
    x, y = _xy(lng, lat, truncate=truncate)
  File "/usr/local/lib/python3.7/site-packages/mercantile/__init__.py", line 307, in _xy
    y = 0.5 - 0.25 * math.log((1.0 + sinlat) / (1.0 - sinlat)) / math.pi
ZeroDivisionError: float division by zero
>>> 
```
Given that the problem arises from Latitude being too large, "ZeroDivisionError" was unhelpful and took me time to unravel the root cause.

I would expect that the errors should be the same, and be helpful, providing the user actionable information regarding how to fix the error.

A fix for this bug would involve calling `mercantile.tile()`  with Lat = 90.0, resulting in a raised exception `mercantile.InvalidLatitudeError: Y can not be computed: lat=90`, consistent with the error raised by -90.
